api/proto/info/info.proto

```diff
@@ -393,6 +394,37 @@ message ZCellularProvider {
   bool roaming = 4;
 }
 
+// Snapshot states
+enum ZSnapshotState {
+  Z_SNAPSHOT_STATE_UNSPECIFIED = 0;
+  // This state is used when a snapshot is in the process of being
+  // created or an error occurred during the first attempt to create it.
+  // (For example, the operation was delayed)
+  Z_SNAPSHOT_STATE_CREATING = 1;
+  // This state is used when the snapshot has been successfully created.
+  Z_SNAPSHOT_STATE_CREATED = 2;
+  // This state is used when the snapshot is pending deletion or
+  // the first deletion attempt was not successful.
+  Z_SNAPSHOT_STATE_DELETING = 3;
+  // This state is used to send information to the controller about a
+  // snapshot that was implicitly deleted after a rollback snapshot
+  // or volume delete command.
+  Z_SNAPSHOT_STATE_IMPLICITLY_DELETED = 4;
+}
+
+// ZInfoSnapshot - Information about snapshot in zfs for zvol
+message ZInfoSnapshot {
+  uint64 creation_time = 1;           // In seconds
+  string uuid = 2;                    // link a command and a response. Is the real name of the snapshot in ZFS
+  string volume_uuid = 3;             // Volume ID of the volume this snapshot belongs to.
+  string display_name = 4;            // Ex: "Tuesday" or creation time (User-friendly name)
+  bool encryption = 5;
+  ZSnapshotState current_state = 6;   // Displays the current state of the snapshot
+  string error_msg = 7;               // Ops error
+  uint32 rollback_cmd_counter = 8;    // Counter for rollback cmd
```

Comment:
Please add "from the DevOpsCmd" to the comment.

One detailed question is in my above example with two snapshots (uuid=1 and uuid=2) followed by a rollback command to 1, then if the controller did not explicitly delete uuid=2 there will be a ZInfoSnapshot for uuid=2 with implicitly deleted. Will that ZInfoSnapshot have a rollback_cmd_counter and timestamp? Or will that only appear for uuid=1 (the snapshow we rolled back to)?

Reply:
> Will that ZInfoSnapshot have a rollback_cmd_counter and timestamp?

I don't think it makes sense since we didn't get a rollback command for its a snapshot, we just found that the snapshot no longer exists in zfs. This will only show up for snapshot with uuid=1

Reply:
I expanded the comment